### ISSUE TITLE
Rename reference `.lite.tsx` file in `mitosis build`, put behind flag

### ISSUE THREAD

**Issue by samijaber** [BuilderIO/mitosis#450]
As per my comment in https://github.com/BuilderIO/mitosis/issues/444#issuecomment-1147455393, this is generating confusion.

Things to do:
- rename the outputted file's extension to `original.lite.tsx`
- Instead of the condition being `target === 'swift' || target === 'svelte'`, it should be a new option e.g. `options.outputOriginal`, which is opt-in (e.g. `false` by default)

The code to change would be here:

https://github.com/BuilderIO/mitosis/blob/a07384c071e80c095001f18154e2d8f7caf91f2c/packages/cli/src/build/build.ts#L262-L267

**Comment by samijaber** [comment]
I corrected a mistake in my description: the reference files are actually the _first version_ of the generated output. In the case of Mitosis's `build` command, there are sometimes additional transpilation steps after generating this initial output (for certain targets like React, React-Native, Soli). 

Therefore it still makes the most sense to keep them in their respective generated output directories.

**Comment by kylecordes** [comment]
As I look back at this, I suspect the best default is to **not** emit this file to the output. That matches the behavior of most other compilers of all kinds, which typically emit intermediate/diagnostic files only on request.

**Comment by samijaber** [comment]
Yeah agreed, that's what I've been thinking throughout this: Get rid of it completely, add an `output: 'ts' | 'js'` flag in the mitosis config. Only output 1 thing based off of it.

**Comment by samijaber** [comment]
Lines up with our separate issues around adding TS output to the CLI commands. The generators currently are starting to output TS, but the CLI has not caught up: it ends up transpiling code down to JS in many cases.
